### PR TITLE
feature/ak-25: Added AWS EBS Snapshot Lifecycle plugin, test cases

### PR DESCRIPTION
**Pull request by AkhtarAmir**

### CODE REVIEW

File: plugins/aws/ec2/ebsSnapshotLifecycle.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+                    helpers.addResult(results, 3,
+                        `Unable to query lifecycle policy: ${helpers.addError(getLifecyclePolicy)}`,
+                        region, policy.PolicyId);
+                    return;
+                }
+
+                if (getLifecyclePolicy.data.Policy.State &&
+                    getLifecyclePolicy.data.Policy.State === 'ENABLED' &&
+                    getLifecyclePolicy.data.Policy.PolicyDetails &&
+                    getLifecyclePolicy.data.Policy.PolicyDetails.TargetTags) {
```
Comment (giorod3) [inline plugins/aws/ec2/ebsSnapshotLifecycle.js:77]: you are looping through this without first checking if it has anything in it
```suggestion
                    getLifecyclePolicy.data.Policy.PolicyDetails.TargetTags && getLifecyclePolicy.data.Policy.PolicyDetails.TargetTags.length) {
```

**Review by giorod3 — CHANGES_REQUESTED** [review]
just the one comment

File: plugins/aws/ec2/ebsSnapshotLifecycle.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+                                    if (passingInstanceTags[tag.Key]) passingInstanceTags[tag.Key].push(tag.Value); 
+                                    else passingInstanceTags[tag.Key] = [tag.Value];
+                                }
+                            });
+                        }
+                    }
+                }
+            });
+
+            describeInstances.data.forEach(instance => {
```
Comment (dipsubha06) [inline plugins/aws/ec2/ebsSnapshotLifecycle.js:100]: Dont we need to check describeInstances.data.length as we are doing for describeVolumes.data.length?

Comment (dipsubha06) [inline plugins/aws/ec2/ebsSnapshotLifecycle.js:100]: Dont we need to check describeInstances.data.length as we are doing for describeVolumes.data.length?

**Review by dipsubha06 — CHANGES_REQUESTED** [review]

**Review by dipsubha06 — APPROVED** [review]

**Review by giorod3 — APPROVED** [review]

**Pull request merged**